examples/schema.rs

```diff
@@ -0,0 +1,137 @@
+#[macro_use]
+extern crate clap;
+use clap::{App, ArgMatches};
+use ion_rs::text::writer::TextWriter;
+use ion_rs::value::reader::element_reader;
+use ion_rs::value::reader::ElementReader;
+use ion_rs::value::writer::{ElementWriter, Format, TextKind};
+use ion_rs::IonType;
+use ion_schema::authority::{DocumentAuthority, FileSystemDocumentAuthority};
+use ion_schema::result::IonSchemaResult;
+use ion_schema::system::SchemaSystem;
+use std::fs;
+use std::path::Path;
+use std::str::from_utf8;
+
+fn main() -> IonSchemaResult<()> {
+    // load the YAML file which has all the CLI commands information stored in it
+    let yaml = load_yaml!("cli.yaml");
+    let matches = App::from_yaml(yaml).get_matches();
+    let (command_name, command_args) = matches.subcommand();
+    let command_args = command_args.unwrap();
+
+    match command_name {
+        "load" => load(command_args)?,
+        "validate" => validate(command_args)?,
+        _ => eprintln!("command name: {} not found", command_name),
```

Comment:
Not super important right now, but it might be nice for users if you included something about how to get help or find the list of valid commands.